internal/config/nsmgr/nsmgr.go

```diff
@@ -187,6 +187,50 @@ func (mgr *NamespaceManager) dirForType(ns NSType) string {
 	return filepath.Join(mgr.namespacesDir, string(ns)+"ns")
 }
 
+// NamespaceFromProcEntry creates a new namespace object from a bind mount from a processes proc entry.
+// The caller is responsible for cleaning up the namespace by calling Namespace.Remove().
+// This function is heavily based on containernetworking ns package found at:
+// https://github.com/containernetworking/plugins/blob/5c3c17164270150467498a32c71436c7cd5501be/pkg/ns/ns.go#L140
+// Credit goes to the CNI authors.
+func (mgr *NamespaceManager) NamespaceFromProcEntry(pid int, nsType NSType) (_ Namespace, retErr error) {
+	pinnedNamespace := uuid.New().String()
+
+	// now create an empty file
+	f, err := os.Create(pinnedNamespace)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating namespace path")
+	}
+	f.Close()
+
+	defer func() {
+		if retErr != nil {
+			if err := os.Remove(pinnedNamespace); err != nil {
+				logrus.Errorf("Failed to remove namespace path %s after failure to pin namespace: %v", pinnedNamespace, err)
```

Comment:
The error message from os.Remove already contains the file name, no need to have it twice.